contracts/global-extensions/GlobalOptimisticAuctionRebalanceExtension.sol

```diff
@@ -0,0 +1,328 @@
+
+        ProductSettings memory settings = productSettings[_setToken];
+        settings.optimisticParams.optimisticOracleV3.settleAndGetAssertionResult(assertionId);
+        
+        address[] memory currentComponents = _setToken.getComponents();
+        
+        require(currentComponents.length == _oldComponents.length, "Mismatch: old and current components length");
+
+        for (uint256 i = 0; i < _oldComponents.length; i++) {
+            require(currentComponents[i] == _oldComponents[i], "Mismatch: old and current components");
+        }
+
+        bytes memory callData = abi.encodeWithSelector(
+            IAuctionRebalanceModuleV1.startRebalance.selector,
+            _setToken,
+            _quoteAsset,
+            _newComponents,
+            _newComponentsAuctionParams,
+            _oldComponentsAuctionParams,
+            _shouldLockSetToken,
+            _rebalanceDuration,
+            _positionMultiplier
+        );
+
+        _invokeManager(_manager((_setToken)), address(auctionModule), callData);
+    }
+
+     // Constructs the claim that will be asserted at the Optimistic Oracle V3.
+    function _constructClaim(bytes32 proposalHash, bytes32 rulesHash) internal pure returns (bytes memory) {
+        return
+            abi.encodePacked(
+                AncillaryData.appendKeyValueBytes32("", PROPOSAL_HASH_KEY, proposalHash),
+                ",",
+                RULES_KEY,
+                ":\"",
+                rulesHash,
+                "\""
+            );
+    }
+
+    /**
+     * @notice Callback function that is called by Optimistic Oracle V3 when an assertion is resolved.
+     * @dev This function does nothing and is only here to satisfy the callback recipient interface.
+     * @param assertionId The identifier of the assertion that was resolved.
+     * @param assertedTruthfully Whether the assertion was resolved as truthful or not.
+     */
+    function assertionResolvedCallback(bytes32 assertionId, bool assertedTruthfully) external {}
+
+    /**
+     * @notice Callback to automatically delete a proposal that was disputed.
+     * @param _assertionId the identifier of the disputed assertion.
+     */
+    function assertionDisputedCallback(bytes32 _assertionId) external {
+        bytes32 proposalHash = proposalHashes[_assertionId];
+        bytes32 assertionId = assertionIds[proposalHash];
+
+        ProductSettings memory settings = productSettings[assertedProducts[assertionId]];
+
+        require(address(settings.optimisticParams.optimisticOracleV3) != address(0), "Invalid oracle address");
+
+
+        // If the sender is the Optimistic Oracle V3, delete the proposal and associated assertionId.
+        if (msg.sender == address(settings.optimisticParams.optimisticOracleV3)) {
+            // Delete the disputed proposal and associated assertionId.
+            delete assertionIds[proposalHash];
+            delete proposalHashes[assertionId];
+
+        } else {
```

Comment:
Can you explain this else block maybe. Who other than the oracle would call this callback ? And in what case / for what purpose ?

Reply:
> Can you explain this else block maybe. Who other than the oracle would call this callback ? And in what case / for what purpose ?

This is applicable in the event of an upgrade between proposal and starting a rebalance. 
> https://github.com/UMAprotocol/protocol/blob/cef7d8880a2168eeb30f54151a8d40b274fbd381/packages/core/contracts/optimistic-governor/implementation/OptimisticGovernor.sol#L359C1-L380C6

It is made much more explicit here through an internal function. Would you prefer that here too?